### DISCUSSION TITLE
What happened to dev branch?

### DISCUSSION THREAD

**Discussion by tajmone** [SicroAtGit/RegEx-Engine#36] [Q&A] [answered]
Ciao @SicroAtGit, I noticed that there are no longer dev branches on the project.

Wouldn't it make sense to keep a `dev` baseline branch for incoming pull requests? just to ensure that any contribution is well tested and polished before integrating it into `main`, and because it allows more freedom in terms of force pushes, rebasing, squashing, etc.

I think that it makes sense to have `dev` as a fix baseline branch, and then allow specialized development branches for developing features that might take longer before they are ready to integrate.

**Comment by SicroAtGit** [comment]
Initially, I had intended:

- to set the `main` branch at GitHub as the default branch. This branch should always have the state of the last tagged release. 
- the `devel` branch was intended for development. Feature branches should be created based on this branch and the feature branches should be merged back into this branch with rebase-merge by pull requests or manually.

The problem was that I could not set GitHub to use a different default branch for forks and pull requests, namely the `devel` branch.

Since that doesn't work and I then wanted to keep the branching model simple, I switched to the branching model where there is only the `main` branch. From this feature branches should be created and there they should be merged again with rebase-merge.

But you are right, I should create a `devel` branch again (will name this branch `develop` this time, is more common). This `develop` branch, I set than as the default branch at GitHub. Before the `develop` branch is merged into `main`, I can then run `git rebase --autosquash` beforehand to squash the commits which have correction commits.

I think it should then only be the original commit author making correction commits to their commits via `git commit --fixup`. I mean, the commits that are then squashed together. This prevents someone else from changing someone else's commit content, otherwise it would be no longer apparent what the original commit author actually contributed with his commit.

**Reply by tajmone** [reply]
I wouldn't change the default branch, because it can lead to problems. Simply, add a `dev` (or `develop` if you fancy verbosity) and ask contributors to set it as the target branch. I know, it sucks because then often people forget to set the target branch to other than the default (i.e. `main` in this case), but then, if it's a simple commit chances are that it could ho straight into `main`, otherwise you'll have to ask to create a new pull-request, since there doesn't seem a way to change the target at present.

I've polished the text of the various README files, but I thought it would be best if I create the PR on a dev branch, in case there are afterthoughts or other changes. But if you prefer, I can just make the PR to `main`.

**Reply by SicroAtGit** [reply]
@tajmone

Yes, I also like it more when a stable branch is the default branch. I find `dev` as the branch name good, quite sufficient.

Regarding pull request was created with the wrong target branch. If I understood correctly, the target branch can now be changed afterwards: [Changing the base branch of a pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request)

> I've polished the text of the various README files, but I thought it would be best if I create the PR on a dev branch, in case there are afterthoughts or other changes. But if you prefer, I can just make the PR to `main`.

I think pull requests that improve documentation and target the `main` branch are fine.

**Comment by SicroAtGit** [comment, marked as answer]
I have created a `dev` branch again.
